### PR TITLE
Err codes

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @jonathandturner

### CODE REVIEW

File: src/librustc_resolve/diagnostics.rs [hunk 1]
```diff
@@ -1270,7 +1270,42 @@ trait Foo {}
...
+```compile_fail,E0530
+static TEST: i32 = 0;
+
+let r: (i32, i32) = (0, 0);
+match r {
+    TEST => {} // error: match bindings cannot shadow statics
+}
+```
+
+To fix this error, just change the binding's name in order to avid shadowing
```
Comment (parir) [inline src/librustc_resolve/diagnostics.rs]: s/avid/avoid/

**Comment by jonathandturner** [comment]
Can't tell for sure, but this might be a legit error:

```
thread 'main' panicked at '
failures:

---- [run-pass] run-pass/issue-36023.rs stdout ----
	
error: test run failed!
status: signal: 11
```

Does this reproduce locally for you @GuillaumeGomez?

**Comment by jonathandturner** [comment]
Looks like maybe not, I just saw the same issue on someone else's PR, too...

**Pull request merged**